packages/core/src/fetch.ts

```diff
@@ -131,12 +131,26 @@ export function buildPlacementsRequest(placements: MozAdsPlacements): AdPlacemen
 /**
  * Maps the ad content from the UAPI response to corresponding placement IDs of given configs.
  *
- * Note: This function makes no guarantee that all given placement IDs will have defined `adContent`.
+ * Note: This function will attempt to use fallback ads where possible if not all content is mapped.
  */
 export function mapResponseToPlacementsWithContent(response: AdResponse, placements: MozAdsPlacements): MozAdsPlacements {
   for (const placementWithContent of Object.values<MozAdsPlacementWithContent>(placements)) {
     const placementId = placementWithContent.placementId
-    placementWithContent.content = response[placementId]?.[0]
+    const contentFromServer = response[placementId]?.[0]
+    if (!contentFromServer) {
+      // If an ad placement is missing from the response, we fill that slot if a single fallback if able
+
+      if (!placementWithContent.fixedSize) {
+        // Without a fixedSize, we cannot fallback to anything, so we give it an empty object to let future steps know loading is complete.
+        placementWithContent.content = {}
+        continue
+      }
+      const fallbackContentType = AdUnitFormatTypeLookup[`${placementWithContent?.fixedSize.width}x${placementWithContent?.fixedSize.height}`]
+      const fallbackContent = fallbackAdContentLookup[fallbackContentType] ?? getFallbackSquareDefault()
+      placementWithContent.content = fallbackContent
+      continue
+    }
+    placementWithContent.content = contentFromServer
```

Comment:
Couldn't this whole new block just be replaced with?:

```
placementWithContent.content = getFallbackAds(placementWithContent)
```

Reply:
Ohhh nevermind, `getFallbackAds` is slightly different in the expected inputs and outputs. I like what you're saying about refactoring some of this fetch/fallback stuff a bit more, but in the meantime I wonder if we could essentially create a new `getFallbackAd` function in fallback.ts that does what you're doing here essentially and then our `getFallbackAds` function can just call that and we at least DRY this up a bit?